configure.ac

```diff
@@ -93,9 +93,68 @@ AC_CANONICAL_HOST
 AC_SUBST(build)
 AC_SUBST(host)
 
+AS_VAR_IF([cross_compiling], [maybe],
+ [AC_MSG_ERROR([Cross compiling required --host=HOST-TUPLE and --build=ARCH])]
+)
+
 # pybuilddir.txt will be created by --generate-posix-vars in the Makefile
 rm -f pybuilddir.txt
 
+dnl cross-compiling needs a freeze_module binary for build platform
+AC_ARG_WITH(
+  [freeze_module],
+  [AS_HELP_STRING([--with-freeze-module=Programs/_freeze_module],
+                  [path to _freeze_module binary for cross compiling (default: self-hosted Programs/_freeze_module binary)])],
+  [
+    AC_MSG_CHECKING([for --with-freeze-module])
+    AS_VAR_IF([cross_compiling], [no], AC_MSG_ERROR([--with-freeze-module only applies to cross compiling]))
+    if test "$with_freeze_module" = yes -o "$with_freeze_module" = no; then
+      AC_MSG_ERROR([invalid --with-freeze-module option: expect path, not "$with_freeze_module"])
+    fi
+    if ! $(command -v "$with_freeze_module" >/dev/null 2>&1); then
+      AC_MSG_ERROR([invalid or missing freeze module binary "$with_freeze_module"])
+    fi
+    FREEZE_MODULE="$with_freeze_module"
+    AC_MSG_RESULT([$FREEZE_MODULE])
+  ], [
+    AS_VAR_IF([cross_compiling], [yes],
+      [AC_MSG_ERROR([Cross compiling requires --with-freeze-module])]
+    )
+    FREEZE_MODULE=Programs/_freeze_module
+  ]
+)
+AC_SUBST([FREEZE_MODULE])
+
+AC_ARG_WITH(
+  [build-python],
+  [AS_HELP_STRING([--with-build-python],
+                  [path to build python binary for cross compiling (default: self-hosted ./python binary)])],
+  [
+    AC_MSG_CHECKING([for --with-build-python])
+    AS_VAR_IF([cross_compiling], [no], AC_MSG_ERROR([--with-build-python only applies to cross compiling]))
+    if test "$with_build_python" = yes -o "$with_build_python" = no; then
+      AC_MSG_ERROR([invalid -with-build-python option: expect path, not "$with_build_python"])
+    fi
+    if ! $(command -v "$with_build_python" >/dev/null 2>&1); then
+      AC_MSG_ERROR([invalid or missing build Python binary "$with_build_python"])
+    fi
+    build_python_ver=$($with_build_python -c "import sys; print(f'{sys.version_info.major}.{sys.version_info.minor}')")
```

Comment:
I think we should check the versions are exactly the same, as before beta two versions of CPython can have different bytecode compilers. This *probably* won't be an issue, but checking the git tag as well would make sure that this isn't a problem. (The commit SHA seems to be `sys._git[-1]`)

Reply:
+1

Reply:
-1

The proposed restrictive test would force us to rebuild the build Python interpreter every time we commit a change or rebase a PR. This would be very annoying and slow down development.

Final patch versions will always have the same bytecode. For development / alpha versions it's ok to have unstable local builds. If something breaks, then we expect developers to do  ``make distclean`` and start over.

Reply:
Yeah that's a good point. Ok then, I will dismiss this.